lumen/ai/vector_store.py

```diff
@@ -378,34 +434,45 @@ def add(self, items: list[dict]) -> list[int]:
             List of assigned IDs for the added items.
         """
         all_texts = []
+        all_text_and_metadatas = []
         all_metadata = []
 
         for item in items:
             text = item["text"]
             metadata = item.get("metadata", {}) or {}
-            chunks = self._chunk_text(text)
+            content_text, text_and_metadata = self._get_content_and_text_and_metadata(
+                text, metadata
+            )
 
-            for chunk in chunks:
+            content_chunks = self._chunk_text(content_text)
+            for chunk in content_chunks:
                 all_texts.append(chunk)
+                all_text_and_metadatas.append(text_and_metadata)
```

Comment:
Seems like we'd be computing embeddings for the same thing a bunch of times?

Reply:
Yeah, so maybe: https://github.com/holoviz/lumen/pull/918